### ISSUE TITLE
 community : Contributor ladder

### ISSUE THREAD

**Issue by naveensrinivasan** [sigstore/sigstore#305]
**Description**

I am opening this to ask if there's a contributor ladder defined for sigstore.
How do I become an org member?

I  would be happy to help do PR's reviews here, hoping to work towards maintainership.

previous contributions - mainly fuzzing sigstore and integrating with oss-fuzz

## PR's in sigstore
1. https://github.com/sigstore/sigstore/pull/214
1. https://github.com/sigstore/sigstore/pull/213
1. https://github.com/sigstore/sigstore/pull/212
1. https://github.com/sigstore/sigstore/pull/197
1. https://github.com/sigstore/sigstore/pull/178
1. https://github.com/sigstore/sigstore/pull/177
1. https://github.com/sigstore/sigstore/pull/173
1. https://github.com/sigstore/sigstore/pull/170
1. https://github.com/sigstore/sigstore/pull/169
1. https://github.com/sigstore/sigstore/pull/168
1. https://github.com/sigstore/sigstore/pull/165
1. https://github.com/sigstore/sigstore/pull/164
1. https://github.com/sigstore/sigstore/pull/160
1. https://github.com/sigstore/sigstore/pull/158
1. https://github.com/sigstore/sigstore/pull/157
1. https://github.com/sigstore/sigstore/pull/148
1. https://github.com/sigstore/sigstore/pull/146
1. https://github.com/sigstore/sigstore/pull/127

## oss-fuzz and actively maintaining the oss-fuzz issues 
1. https://github.com/google/oss-fuzz/pull/6890
1. https://github.com/google/oss-fuzz/pull/6927
1. https://github.com/google/oss-fuzz/pull/6964

## Issues in sigstore
https://github.com/sigstore/sigstore/issues?q=is%3Aissue+author%3Anaveensrinivasan

## PR's in cosign
1. https://github.com/sigstore/cosign/pull/1141
1. https://github.com/sigstore/cosign/pull/1020
1. https://github.com/sigstore/cosign/pull/1001
1. https://github.com/sigstore/cosign/pull/971
1. https://github.com/sigstore/cosign/pull/968
1. https://github.com/sigstore/cosign/pull/944
1. https://github.com/sigstore/cosign/pull/124
1. https://github.com/sigstore/cosign/pull/121
1. https://github.com/sigstore/cosign/pull/120
1. https://github.com/sigstore/cosign/pull/119

## Issues in cosign
https://github.com/sigstore/cosign/issues?q=is%3Aissue+author%3Anaveensrinivasan+

## PR's rekor

https://github.com/sigstore/rekor/pulls?q=author%3Anaveensrinivasan

## Issues in rekor 

https://github.com/sigstore/rekor/issues?q=author%3Anaveensrinivasan

cc @lukehinds @dlorenc @bobcallaway

**Comment by lukehinds** [comment]
We don't have anything defined as each project has autonomy to manage its own maintainers (codeowners).

As a general guide, I myself view a maintainer as someone who regularly helps review code, finds and resolves bugs and adds features. A good candidate is someone who has a consistent presence in the project.

I hope that helps and sorry for not being more specific. Currently a lot of your contributions (of a varied variety) are towards cosign, so that looks like a good trajectory towards being a maintainer.

**Comment by naveensrinivasan** [comment]
Good to know. Thanks,I would like and being interested in becoming a maintainer now.

**Comment by dlorenc** [comment]
I kind of miss having something like peribolos to manage permissions across an org, but don't really want to have to setup prow just for that. @cpanato do you know of any way to do that easier?

**Comment by cpanato** [comment]
> I kind of miss having something like peribolos to manage permissions across an org, but don't really want to have to setup prow just for that. @cpanato do you know of any way to do that easier?

I did this: https://github.com/cpanato/pulumi-github-sync and implemented it at mattermost, it is working just fine and easy to deploy and get those up and running

**Comment by cpanato** [comment]
yes! can we create a new repo that I can push the code and set up? can we don't need to connect to the pulumi UI, we can use GCP Storage as the state storage

**Comment by dlorenc** [comment]
Thanks! 

I realize this doesn't solve the actual contributor ladder problem, but it at least makes it so the mechanics of joining as a contributor are clearly defined and transparent. We can figure out the hard part next :)

**Comment by naveensrinivasan** [comment]
> yes! can we create a new repo that I can push the code and set up? can we don't need to connect to the pulumi UI, we can use GCP Storage as the state storage
> 
> 
> 
> 
> 
> I will ping you in the sigstore slack


Pulumi is cool. But the stack would be in a personal Pulumi account and cannot be shared amongst members unless a paid account. 

How is this going to be handled?

**Comment by cpanato** [comment]
we will not use the pulumi account, we will store the state in the gcp storage, we will miss some nice features, but that will work fine for us

**Comment by cpanato** [comment]
i was speaking with Dan to re-use some docs/process from he k8s :D

**Comment by justaugustus** [comment]
> i was speaking with Dan to re-use some docs/process from he k8s :D

Feel free to tag me for reviews, as this is something I'm planning to do for scorecard and friends and hopefully something lightweight/generic enough to use for all of OpenSSF.

**Comment by justaugustus** [comment]
w.r.t. contributing guides, I'm in the midst of rewriting the one for Kubernetes SIG Release, which I think is coming along pretty nicely: https://github.com/kubernetes/sig-release/pull/1862

**Comment by justaugustus** [comment]
In terms of actually operationalizing this request, what I'd likely do is approve, add @naveensrinivasan to the org, create a team for triagers/reviewers, and give that team `triage` permissions on the requisite repos.

[`triage`](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) is a nice middleground between no/read access and destructive actions.

**Comment by dlorenc** [comment]
> In terms of actually operationalizing this request, what I'd likely do is approve, add @naveensrinivasan to the org, create a team for triagers/reviewers, and give that team `triage` permissions on the requisite repos.

+1, approve!

**Comment by lukehinds** [comment]
> In terms of actually operationalizing this request, what I'd likely do is approve, add @naveensrinivasan to the org, create a team for triagers/reviewers, and give that team `triage` permissions on the requisite repos.
> 
> [`triage`](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) is a nice middleground between no/read access and destructive actions.

I like this as well. That way contributors can first go to triage and we bring on new codeowners when there is a real need (lack of reviewers who can merge). This way we can have people recognised and have some ability to help out with housekeeping, but we don't end up with large codeowner grants.

**Comment by naveensrinivasan** [comment]
> > In terms of actually operationalizing this request, what I'd likely do is approve, add @naveensrinivasan to the org, create a team for triagers/reviewers, and give that team `triage` permissions on the requisite repos.
> > [`triage`](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) is a nice middleground between no/read access and destructive actions.
> 
> I like this as well. That way contributors can first go to triage and we bring on new codeowners when there is a real need (lack of reviewers who can merge). This way we can have people recognised and have some ability to help out with housekeeping, but we don't end up with large codeowner grants.

Friendly ping!

**Comment by dlorenc** [comment]
Invite sent!

**Comment by dlorenc** [comment]
Reopening to track the rest of the ladder process :)

**Comment by lukehinds** [comment]
Does this mean @naveensrinivasan (or whoever we add) has triage over all projects in the org, or just sigstore/sigstore?

**Comment by dlorenc** [comment]
Just sigstore/sigstore

**Comment by cpanato** [comment]
I'm going to work on the docs this week

**Comment by lukehinds** [comment]
closing as now tracked in https://github.com/sigstore/community/pull/54
